pkg/reconciler/v1alpha1/autoscaling/kpa_scaler.go

```diff
@@ -127,6 +128,20 @@ func (ks *kpaScaler) Scale(ctx context.Context, kpa *kpa.PodAutoscaler, desiredS
 	}
 	currentScale := scl.Spec.Replicas
 
+	var serviceLabel string
+	var configLabel string
+	if kpa.Labels != nil {
+		serviceLabel = kpa.Labels[serving.ServiceLabelKey]
+		configLabel = kpa.Labels[serving.ConfigurationLabelKey]
```

Comment:
I don't like two things about this:
1. the knowledge of `serving` feels like a layering violation
1. it can panic

I expect that this WOULD panic if we just used a Route/Configuration and not a Service, which is scary.

Reply:
@mattmoor can you elaborate? In which way would it panic?

Reply:
My mistake:
> If the requested key doesn't exist, we get the value type's zero value

... in this case, the empty string.

The pattern of `val, ok := m[key]` is too ingrained in my head, and close enough to type assertions (which do panic) that I confused them.

Reply:
@mattmoor any suggestions to work around the layering violation?

Reply:
Can you open an issue to rework the interface of `autoscaler.NewStatsReporter`?  Perhaps it should take the label `map[string]string` instead, but it's current signature forces its own layering violation on the caller, so you don't have a lot of options here. :)